(n: int)! => int {
  : r : int := 1;
  if n > 0 
    // RMS: bug here. Exception thrown because N isn't found in the closed scope.
	r = n * {n - 1}!;

  r
}

entrypoint => void {
  : n : int := read int from console;

  print n!
}